### PR TITLE
Feature/storage credential option encryption

### PR DESCRIPTION
**Pull request by jianwel**
refactored the encryptor handler out of the Wallet, and added the encrpt/decrpt for storage credential option

### CODE REVIEW

**Comment by jamesfneyer** [comment]
Could you add yourself as a contributor on the Readme? Would be at around line 1098

File: src/entity/Contract.ts [hunk 1]
```diff
@@ -242,7 +242,7 @@ export class Version extends BaseEntity {
                     logger.debug(`\tLocal Test Net Account: ${account}`);
                     driver
                         .deploy({ data: this.bytecode })
-                        .send({ from: account, gasLimit: 8000000 }, (err, tx_hash) => {
+                        .send({ from: account, gasLimit: 134217728 }, (err, tx_hash) => {
```
Comment (mlclay) [inline src/entity/Contract.ts]: This was part of our debug session for the initial network issue you had.  Please remove this change.

File: src/entity/StorageCredential.ts [hunk 2]
```diff
@@ -37,15 +38,18 @@ export class StorageCredential extends BaseEntity {
...
 
-    static generate_entity(attrs: StorageCredentialAttrs) {
+    static async generate_entity(attrs: StorageCredentialAttrs) {
         const credentials = new StorageCredential();
         credentials.title = attrs.title;
         credentials.driver_type = attrs.driver_type;
         credentials.base_path = attrs.base_path || './';
-        credentials.options = attrs.options || {};
+
+        //credentials.options = attrs.options || {};
```
Comment (mlclay) [inline src/entity/StorageCredential.ts]: Please remove commented/old code

File: src/entity/StorageCredential.ts [hunk 3]
```diff
@@ -100,15 +104,20 @@ export class StorageCredential extends BaseEntity {
...
+            this.options = { EncryptedJson: encryptString };
         }
 
         await this.save();
     }
 
-    getDriverOptions() {
+    async getDriverOptions() {
+        const encryptString = this.options['EncryptedJson'];
+        const decrptedOptionString = await EncryptorContainer.defaultEncryptor.decrypt(encryptString);
```
Comment (mlclay) [inline src/entity/StorageCredential.ts]: Missing `y` in `decrypted`

File: src/entity/migration/1558380631422-StorageCredentialEncryptionOptions.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import { AwsPrivateKeyDBFieldEncryption } from "../../shipchain/AwsPrivateKeyDBFieldEncryption";
+import { PrivateKeyDBFieldEncryption } from "../encryption/PrivateKeyDBFieldEncryption";
+import { DBFieldEncryption } from "../encryption/DBFieldEncryption";
+import { StorageCredential } from "../StorageCredential";
+
+export class StorageCredentialEncryptionOptions1558380631422 implements MigrationInterface {
+
+    private async getEncyptHandler() : Promise<DBFieldEncryption> {
+        const ENV = process.env.ENV || 'LOCAL';
+        if (ENV === "DEV" || ENV === "STAGE" || ENV === "PROD") {
```
Comment (mlclay) [inline src/entity/migration/1558380631422-StorageCredentialEncryptionOptions.ts]: You are missing `DEMO` environment here.

File: src/shipchain/AwsPrivateKeyDBFieldEncryption.ts [hunk 5]
```diff
@@ -15,9 +15,9 @@
  */
 
 import { PrivateKeyDBFieldEncryption } from '../entity/encryption/PrivateKeyDBFieldEncryption';
-import { DBFieldEncryption } from '../entity/Wallet';
 import EthCrypto from 'eth-crypto';
 import { getAwsSecret } from './utils';
+import { DBFieldEncryption } from '../entity/encryption/DBFieldEncryption';
```
Comment (mlclay) [inline src/shipchain/AwsPrivateKeyDBFieldEncryption.ts]: In general there is no strict rule on the placements of imports; however, I prefer if we keep common imports near each other.  Move this line back below the other `from '../entity/...` line above.

File: src/shipchain/__tests__/testShipChainEncryptorContainer.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+require('../../__tests__/testLoggingConfig');
+
+import 'mocha'
+
+import '../ShipChainEncryptorContainer'
```
Comment (mlclay) [inline src/shipchain/__tests__/testShipChainEncryptorContainer.ts]: You are importing this twice.  Remove this line and keep the one below where we import the specific exported class from the module.

**Pull request merged**